### PR TITLE
octanifies new-directory-user component

### PR DESCRIPTION
**Pull request by stopfstedt**

### CODE REVIEW

File: app/components/new-directory-user.hbs [hunk 1]
```diff
@@ -1,292 +1,320 @@
 {{! template-lint-disable no-attrs-in-components }}
```
Comment (jrjohnson) [inline app/components/new-directory-user.hbs:1]: Are these still needed? Can the need for them be removed in this refactor?

File: app/components/new-directory-user.js [hunk 2]
```diff
@@ -1,127 +1,208 @@
...
+      if (currentCohort) {
+        return currentCohort;
+      }
+    }
+
+    return this.currentSchoolCohorts.lastObject;
+  }
+
+  async loadSchools() {
+    const schools = await this.store.findAll('school', { reload: true });
```
Comment (jrjohnson) [inline app/components/new-directory-user.js:78]: I don't think the `reload: true` is necessary anymore. This seems like a workaround we may have needed in the past.

File: app/mixins/newuser.js [hunk 3]
```diff
@@ -1,190 +0,0 @@
-import Mixin from '@ember/object/mixin';
```
Comment (jrjohnson) [inline app/mixins/newuser.js:1]: 🎉

File: app/components/new-directory-user.js [hunk 4]
```diff
@@ -1,127 +1,208 @@
...
+    this.schoolId = id;
+  }
+
+  @action
+  setPrimaryCohort(id) {
+    this.primaryCohortId = id;
+  }
+
+  @action
+  async keyboard(event) {
```
Comment (jrjohnson) [inline app/components/new-directory-user.js:173]: I'm not understanding why this method is async (and why the calls to `perform()` need to be awaited here and not just proceed after the keypress.

File: app/components/new-directory-user.js [hunk 5]
```diff
@@ -132,10 +213,57 @@ export default Component.extend(NewUser, Validations, {
...
+      'phone',
+      'username',
+      'password',
+    ]);
+    const isValid = yield this.isValid();
+    if (!isValid) {
+      return false;
+    }
+    const roles = yield this.store.findAll('user-role');
+    const primaryCohort = yield this.bestSelectedCohort;
```
Comment (jrjohnson) [inline app/components/new-directory-user.js:240]: `this.bestSelectedCohort` is a getter and not async so doesn't need to be `yield`ed.

**Review by jrjohnson — CHANGES_REQUESTED** [review]

File: app/components/new-directory-user.js (hunk 5 above)
Comment (stopfstedt) [reply app/components/new-directory-user.js]: :heavy_check_mark:

File: app/components/new-directory-user.js (hunk 4 above)
Comment (stopfstedt) [reply app/components/new-directory-user.js]: :heavy_check_mark:

File: app/components/new-directory-user.js (hunk 2 above)
Comment (stopfstedt) [reply app/components/new-directory-user.js]: :heavy_check_mark:

File: app/components/new-directory-user.hbs (hunk 1 above)
Comment (stopfstedt) [reply app/components/new-directory-user.hbs]: :heavy_check_mark:

**Review by jrjohnson — APPROVED** [review]

**Pull request merged**